### PR TITLE
Add a `Create a DANDI Account` page

### LINKED ISSUE #92: refactor docs for creating an account
We currently have docs for creating an account [here](https://www.dandiarchive.org/handbook/13_upload/#create-an-account-on-dandi) within the section for creating and uploading Dandisets. This is not ideal, because a user may want to create an account in order to access DANDI Hub to analyze existing data, and such a user might find it difficult to find this section. I think it would be better to have a separate section on signing up for DANDI that is linked to by the section for creating and uploading Dandisets as well as the section on the DANDI Hub. This section should also explain when a DANDI login is necessary and when it is not (i.e. if all you want to do is download data)

### PR DESCRIPTION
**Pull request by kabilar**
- [x] Fix #92

### CODE REVIEW

File: docs/16_account.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Without a DANDI account, you can search, view, and download files.
+With a DANDI account, you can additionally create Dandisets and access the DANDI Hub to analyze existing data.
+
+## Instructions
+
+1. To create a DANDI account, first [create a GitHub account](https://github.com/) if you don't have one.
+1. Using your GitHub account, register for a DANDI account by selecting the `LOG IN WITH GITHUB` button on the [DANDI homepage](https://dandiarchive.org).
+1. You will receive an email acknowledging activation of your account within 24
+hours, after which you can log in to DANDI using GitHub by clicking the 
+`LOG IN WITH GITHUB` button.
```
Comment (yarikoptic) [inline docs/16_account.md:13]: for consistency and to set expectations:

```suggestion
1. You will receive an email acknowledging that your request for an account will be reviewed within 24 hours:
  - **note**: requests from a new GitHub accounts and from emails not withing `.edu` domain might take longer to review and more likely to be rejected, especially if there are no plans described to upload data to the archive
1. If your request for an account approved, you would be able to log in to DANDI using GitHub by clicking the `LOG IN WITH GITHUB` button.
```

File: docs/16_account.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+A DANDI account is only required for specific features.
+Without a DANDI account, you can search, view, and download files.
+With a DANDI account, you can additionally create Dandisets and access the DANDI Hub to analyze existing data.
+
+## Instructions
+
+1. To create a DANDI account, first [create a GitHub account](https://github.com/) if you don't have one.
+1. Using your GitHub account, register for a DANDI account by selecting the `LOG IN WITH GITHUB` button on the [DANDI homepage](https://dandiarchive.org).
+1. You will receive an email acknowledging that your request for an account will be reviewed within 24 hours:
```
Comment (kabilar) [inline docs/16_account.md:11]:
```suggestion
1. You will receive an email acknowledging that your request for an account will be reviewed within 24 hours.
```

File: docs/16_account.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+Without a DANDI account, you can search, view, and download files.
+With a DANDI account, you can additionally create Dandisets and access the DANDI Hub to analyze existing data.
+
+## Instructions
+
+1. To create a DANDI account, first [create a GitHub account](https://github.com/) if you don't have one.
+1. Using your GitHub account, register for a DANDI account by selecting the `LOG IN WITH GITHUB` button on the [DANDI homepage](https://dandiarchive.org).
+1. You will receive an email acknowledging that your request for an account will be reviewed within 24 hours.
+  - **note**: requests from a new GitHub accounts and from emails not withing `.edu` domain might take longer to review and more likely to be rejected, especially if there are no plans described to upload data to the archive
+1. If your request for an account approved, you would be able to log in to DANDI using GitHub by clicking the `LOG IN WITH GITHUB` button.
```
Comment (kabilar) [inline docs/16_account.md:13]:
```suggestion
1. If your request for an account is approved, you will be able to log in to DANDI using GitHub by clicking the `LOG IN WITH GITHUB` button.
```

File: docs/16_account.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+A DANDI account is only required for specific features.
+Without a DANDI account, you can search, view, and download files.
+With a DANDI account, you can additionally create Dandisets and access the DANDI Hub to analyze existing data.
+
+## Instructions
+
+1. To create a DANDI account, first [create a GitHub account](https://github.com/) if you don't have one.
+1. Using your GitHub account, register for a DANDI account by selecting the `LOG IN WITH GITHUB` button on the [DANDI homepage](https://dandiarchive.org).
+1. You will receive an email acknowledging that your request for an account will be reviewed within 24 hours.
+  - **note**: requests from a new GitHub accounts and from emails not withing `.edu` domain might take longer to review and more likely to be rejected, especially if there are no plans described to upload data to the archive
```
Comment (kabilar) [inline docs/16_account.md:12]:
```suggestion
  - **note**: Requests from new GitHub accounts and from emails that are not an `.edu` domain might take longer to review and are more likely to be rejected, especially if there are no plans described to upload data to the archive.
```

**Review by bendichter — APPROVED** [review]

**Comment by yarikoptic** [comment]
Approved, looks good to me too -- let's proceed!

**Pull request merged**